pyfar/classes/transmission_matrix.py

```diff
@@ -191,10 +193,13 @@ def from_abcd(cls, A, B, C, D, frequencies = None):
             raise ValueError(
                         "If using FrequencyData objects, all matrix entries "
                         "A, B, C, D, must be FrequencyData objects.")
-
         if frequencies is None:
             raise ValueError("'frequencies' must be specified if not using "
                              "'FrequencyData' objects as input.")
+        
+        # Convert all parameters of type Number to 1d np arrays for further computation
+        A, B, C, D = [np.atleast_1d(np.asarray(param)) if isinstance(param, Number) else param for param in (A, B, C, D)]
```

Comment:
i think you can skip the if check, convert an array into an array would not change anything.